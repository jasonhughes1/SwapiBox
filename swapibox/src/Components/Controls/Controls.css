.button {
  margin: 5px;
  height: 50px;
  border-radius: 10px;
  width: 180px;
  cursor: pointer;
  font-family: inherit;
  font-size: 20px;
  outline: none;
  background-color: none; }

.inactive {
  background-color: green; }

.active {
  background-color: blue; }

button:hover {
  background-color: #F50E1D; }
