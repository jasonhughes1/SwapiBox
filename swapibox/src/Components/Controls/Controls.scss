button {
  margin: 5px;
  height: 50px;
  border-radius: 10px;
  width: 180px;
  border: 2px solid #A0A469;
  cursor: pointer;
  font-family: inherit;
  font-size: 20px;
  outline: none;
}

button:hover {
  background-color: yellow;
}
