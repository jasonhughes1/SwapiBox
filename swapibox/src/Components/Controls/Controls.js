import React from 'react';
import './Controls.css';
import PropTypes from 'prop-types';


const Controls = ({ buttonText, changeCards, num, className }) => {
  return (
    <div className={`${className}`}>
      <button onClick={() => (changeCards(num))}>{buttonText}</button>
    </div>
  );
};

Controls.propTypes = {
  buttonText: PropTypes.string,
  changeCards: PropTypes.func,
  className: PropTypes.string,
  num: PropTypes.number
};

export default Controls;
