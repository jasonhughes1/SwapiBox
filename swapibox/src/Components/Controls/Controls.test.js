import React from 'react';
import Controls from './Controls.js';
import { shallow } from 'enzyme';


const mockChangeCards = jest.fn();
let renderedControls;

describe('Controls Tests', () => {
  beforeEach(() => {
    renderedControls = shallow(
      <Controls
        changeCards={mockChangeCards}
      />);
  });

  it('should render correctly', () => {
    expect(renderedControls).toBeDefined();
  });

  it('should render a button', () => {
    expect(renderedControls.find('button').length).toEqual(1);
  });


  it('should run the chooseCategory function when a button is clicked', () => {
    expect(mockChangeCards.mock.calls.length).toEqual(0);

    renderedControls.find('button').first().simulate('click');

    expect(mockChangeCards.mock.calls.length).toEqual(1);

    renderedControls.find('button').last().simulate('click');

    expect(mockChangeCards.mock.calls.length).toEqual(2);
  });

  it('should match snapshot', () => {
    expect(renderedControls).toMatchSnapshot();
  })
});
