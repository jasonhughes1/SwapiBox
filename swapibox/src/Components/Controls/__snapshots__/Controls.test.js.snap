// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls
    changeCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="undefinedactive"
        onClick={[Function]}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "undefinedactive",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="undefinedactive"
          onClick={[Function]}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "undefinedactive",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
